### DISCUSSION TITLE
[Feature] Improved performance for web previous/next picture - preload next image caching

### DISCUSSION THREAD

**Discussion by LesKent** [immich-app/immich#8149] [Feature Request]
### The feature

When using Immich via a browser, when you are viewing images there appears to be obvious caching of images going on, such that if you select the previous or next picture for one you've already seen, the next image displays instantly. If the next image is not cached, it displays a spinning circle until it is loaded. In my case, typically about a second or two. Given that most people spend some time looking at the current picture before they select the next one, it would great if it is easy to pre-load the next and previous pictures caches in the background, so then choosing the next picture in either direction will display instantly.  I believe this would make the web user interface when browsing pictures feel much faster and fluid. This would also better match the mobile experience where scrolling left/right feels almost instantaneous all the time.  

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#7920
